online-shopping-website/frontend/src/Components/Products/ProductDetails.js

```diff
@@ -128,16 +127,25 @@ const ProductButtons = (props) => {
                 </Button>
                 <input className="inputne" disabled={true} value={quantity}/>
                 <Button className="QuantityButtons-Shared GreenButtonContained" variant="contained"
-                        disabled={quantity === 10}
+                        disabled={((productInCart ? productInCart.quantity : 0) + quantity) >= props.product.totalQuantity}
                         onClick={IncrementItem}>
                     <AddIcon/>
                 </Button>
             </Stack>
-            <h5 className="ProductDetails-ProductLimitText">Limit of 10 items per product in cart.</h5>
+            <h5 className="ProductDetails-ProductLimitText">There {props.product.totalQuantity > 1 ? 'are' : 'is'} {props.product.totalQuantity} in stock.</h5>
+            {
+                // If product is in the user's cart, show quantity of item in cart
+                productInCart && 
+                <h5 className="ProductDetails-ProductLimitText">You currently have {productInCart.quantity} in your cart.</h5>
+            }
             <Button className="ProductDetails-CartButton GreenButtonContained"
                     variant="contained"
                     endIcon={<AddShoppingCartIcon/>}
-                    disabled={false} //FIX ME: Disable button if user is a seller or admin
+                    // Can't add to cart if logged out, or non-customer, or adding to cart would surpass total quantity
+                    disabled={
+                        !cookies.user 
```

Comment:
Damn I didn't think about adding the condition directly on the disabled prop. I didn't know that was a thing

Reply:
Yeah I opted to leave the logic in there so it's easier and more readable. Originally we had logic in the decrease and increase functions but this made more sense it my eyes since it disables it before the option is even there!
Pretty cool way to do it